extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,886 @@
+  filterSort: RepositoriesFilterSortStateWithIds = defaultFilterSortState,
+  credentials: Credentials,
+  app: App,
+  cliServer: CodeQLCliServer,
+): Promise<void> {
+  await withProgress(
+    async (progress: ProgressCallback) => {
+      // Get the path to the local autofix installation.
+      const localAutofixPath = await findLocalAutofix();
+
+      // Get the variant analysis with the given id.
+      const variantAnalysis =
+        variantAnalysisManager.tryGetVariantAnalysis(variantAnalysisId);
+      if (!variantAnalysis) {
+        throw new Error(`No variant analysis with id: ${variantAnalysisId}`);
+      }
+
+      // Generate the query help and output it to the override directory.
+      await overrideQueryHelp(variantAnalysis, cliServer, localAutofixPath);
+
+      // Get the full names (nwos) of the selected repositories.
+      const selectedRepoNames = getSelectedRepositoryNames(
+        variantAnalysis,
+        filterSort,
+      );
+
+      // Get storage paths for the autofix results.
+      const {
+        variantAnalysisIdStoragePath,
+        sourceRootsStoragePath,
+        autofixOutputStoragePath,
+      } = await getStoragePaths(variantAnalysisManager, variantAnalysisId);
+
+      // Process the selected repositories:
+      //  Get sarif
+      //  Download source root
+      //  Run autofix and output results
+      progress(
+        progressUpdate(
+          1,
+          2,
+          `Processing ${pluralize(selectedRepoNames.length, "repository", "repositories")}`,
+        ),
+      );
+      const outputTextFiles = await processSelectedRepositories(
+        variantAnalysisResultsManager,
+        selectedRepoNames,
+        variantAnalysisIdStoragePath,
+        sourceRootsStoragePath,
+        autofixOutputStoragePath,
+        localAutofixPath,
+        credentials,
+      );
+
+      // Output results from all repos to a combined markdown file.
+      progress(progressUpdate(2, 2, `Finalizing autofix results`));
+      const combinedOutputMarkdownFile = join(
+        autofixOutputStoragePath,
+        "autofix-output.md",
+      );
+      await mergeFiles(outputTextFiles, combinedOutputMarkdownFile, false);
+
+      // Open the combined markdown file.
+      await tryOpenExternalFile(app.commands, combinedOutputMarkdownFile);
+    },
+    {
+      title: "Generating Autofixes",
+      cancellable: false, // not cancellable for now
+    },
+  );
+}
+
+/**
+ * Finds the local autofix installation path from the `codeQL.autofix.path` setting.
+ * Throws an error if the path is not set or does not exist.
+ * @returns An object containing the local autofix path.
+ * @throws Error if the `codeQL.autofix.path` setting is not set or the path does not exist.
+ */
+async function findLocalAutofix(): Promise<string> {
+  const localAutofixPath = getAutofixPath();
+  if (!localAutofixPath) {
+    throw new Error(
+      "Path to local autofix installation not found. Internal GitHub access required.",
+    );
```

Comment:
Would it be useful to add the name of the config setting here (either hard-coded or using `AUTOFIX_PATH.qualifiedName`)?

Reply:
Done in https://github.com/github/vscode-codeql/pull/4065/commits/bf6d8e72b871f3068c1c89465d2ad527389d90de.